### ISSUE TITLE
Add chat comments to a timed queue

### ISSUE THREAD

**Issue by aalkz** [aaronpk/live-chat-overlay#30]
I think this would be really useful for a lot of people like me who want to highlight a lot of comments along the stream but don't want to manually be worrying about the timing, having to turn them on or off. A lot of times I've left a comment on for too long and I have to constantly be aware of what's on screen and keeping a "timer" in my head asking myself if the comment has been on long enough for people to have read it.

This is the behavior I imagine:

- Turn option on in settings (not on by default) (alternative: add timer button to each comment so it functions independently)
- Add chat comments to queue while the stream is going on
- Shows one comment at a time for 5 seconds (maybe configured option later?) one at a time until the queue is clear
- Option to clear queue
- Extra: Pin a comment so it's on regardless of the queue

This is a great extension, by the way. I've been using it a lot lately, great job Aaron! :)

**Comment by aaronpk** [comment]
This is a cool idea! So I'm assuming in this case you're treating the chat mainly as a passive thing displayed on screen, but you want to show only selected comments (unlike a lot of other solutions which show every comment on screen)?

**Comment by aalkz** [comment]
Yes, that's exactly it! Sometimes I do use it to highlight something in sync with what I'm saying (eg. QA section), but not always. To give a little bit of context, the streams I run are guided collective prayers and people really like to be mentioned by the hosts, but it's impossible to mention every single one, so the next best thing is for their comment/prayer to appear on screen while the main prayer is going on. I can imagine this being useful for anyone though, as a type of nuanced comment curation by the streamer themselves, rather than just showing the whole chat box as you've correctly pointed out. I stay away from those solutions because I don't want the occasional troll comment be burned in the video forever after the stream is done.

**Comment by copperdome** [comment]
I was just about to make a request when I saw this. I have been using this as a producer who is on screen as well. Manually highlighting is a challenge but I love the ability to do it so I can call out questions and directly answer then. The biggest downside is that they don't go away on their own and the only way I can get rid of them is either clicking another message (which animates between the two) or clicking clear (which makes them abruptly disappear).  I was going to ask for animation on clear... but having them on a timer would remove the need to clear out each message every time.. Even better would be to have the timer auto adjust for message length to make sure longer messages are readable.

**Comment by aaronpk** [comment]
I like the idea of the message length factoring in to how long the messages show on screen! Great idea!

I think this would make an excellent addition to the extension, and could be done fairly easily now that the display of the messages is moved to a standalone page. I will try to get to this in the near future if nobody else tackles it first!

**Comment by MadisonHurler** [comment]
Have you made any progress on this? I would like to be able to instruct the overlay to display the newest comment upon its arrival and display it for 30 seconds and then have it disappear.
